src/OmniSharp.Roslyn/BufferManager.cs

```diff
@@ -128,37 +131,54 @@ private bool TryAddTransientDocument(string fileName, string fileContent)
             var projects = FindProjectsByFileName(fileName);
             if (!projects.Any())
             {
+                //todo: deal with the deletion of the misc files
+                if (fileName.EndsWith(".cs") && _workspace.TryAddMiscellaneousDocument(fileName, LanguageNames.CSharp) != null)
+                {
+                    _fileSystemWatcher.Watch(fileName, OnMiscFileChanged);
```

Comment:
Can you store the OnMiscFileChanged delegate in a field so we don't have to create one instance for every file we watch?